src/snowflake/cli/plugins/snowpark/package/commands.py

```diff
@@ -203,58 +176,60 @@ def package_create(
         package = Requirement.parse(name)
-        if anaconda.is_package_available(
-            package, skip_version_check=skip_version_check
-        ):
+        download_result = download_unavailable_packages(
+            requirements=[package],
+            target_dir=packages_dir,
+            ignore_anaconda=ignore_anaconda,
+            allow_shared_libraries=allow_shared_libraries_yesnoask,
+            skip_version_check=skip_version_check,
+            pip_index_url=index_url,
+        )
+        if not download_result.succeeded:
+            raise ClickException(download_result.error_message)
+
+        # check if package was detected as available
+        package_available_in_conda = any(
+            p.line == package.line
+            for p in download_result.packages_available_in_anaconda
+        )
+        if package_available_in_conda:
             return MessageResult(
                 f"Package {name} is already available in Snowflake Anaconda Channel."
             )
 
-    packages_dir = SecurePath(".packages")
-    packages_are_downloaded, dependencies = download_packages(
-        anaconda=anaconda,
-        ignore_anaconda=ignore_anaconda,
-        package_name=name,
-        requirements_file=None,
-        packages_dir=packages_dir,
-        index_url=index_url,
-        allow_shared_libraries=allow_shared_libraries_yesnoask,
-        skip_version_check=skip_version_check,
-    )
-
-    if not packages_are_downloaded:
-        return MessageResult(
-            dedent(
-                f"""
-                Cannot create package for {name}. Please check the package name
-                or try again with --allow-shared-libraries option.
-                """
-            )
-        )
-
-    # The package is not in anaconda, so we have to pack it
-    zip_file = create_packages_zip(name)
-    message = dedent(
-        f"""
-        Package {zip_file} created. You can now upload it to a stage using
-        snow snowpark package upload -f {zip_file} -s <stage-name>`
-        and reference it in your procedure or function.
-        Remember to add it to imports in the procedure or function definition.
-        """
-    )
-    if dependencies.snowflake:
-        message += dedent(
+        # The package is not in anaconda, so we have to pack it
+        log.info("Checking to see if packages have shared (.so/.dll) libraries...")
+        if detect_and_log_shared_libraries(download_result.downloaded_packages_details):
+            if not resolve_allow_shared_libraries_yes_no_ask(
+                allow_shared_libraries_yesnoask
+            ):
+                raise ClickException(
+                    "Some packages contain shared (.so/.dll) libraries. "
+                    "Try again with --allow-shared-libraries."
+                )
+
+        zip_file = f"{get_package_name(name)}.zip"
+        zip_dir(dest_zip=Path(zip_file), source=packages_dir.path)
+        message = dedent(
```

Comment:
As discussed in the other thread we should split download from checking for native libraires. In this way code is easier to compose and test.

Reply:
This is already a separate part of command code (or I don't see something :D)